packages/common-utils/src/renderChartConfig.ts

```diff
@@ -816,36 +816,80 @@ function translateMetricChartConfig(
       whereLanguage: 'sql',
     };
   } else if (metricType === MetricsDataType.Sum && metricName) {
+    const time_bucket_col = '`__hdx_time_bucket2`';
+    const value_high_col = '`__hdx_value_high`';
+    const value_high_prev_col = '`__hdx_value_high_prev`';
+    const time_expr = timeBucketExpr({
+      interval: chartConfig.granularity || 'auto',
+      timestampValueExpression:
+        chartConfig.timestampValueExpression || 'TimeUnix',
+      dateRange: chartConfig.dateRange,
+      alias: '__hdx_time_bucket2',
+    });
     return {
       ...restChartConfig,
       with: [
         {
-          name: 'RawSum',
-          sql: chSql`SELECT *,
-               any(Value) OVER (ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING) AS PrevValue,
-               any(AttributesHash) OVER (ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING) AS PrevAttributesHash,
-               IF(AggregationTemporality = 1,
-                  Value,IF(Value - PrevValue < 0 AND AttributesHash = PrevAttributesHash, Value,
-                      IF(AttributesHash != PrevAttributesHash, 0, Value - PrevValue))) as Rate
-            FROM (
-                SELECT *, 
-                       cityHash64(mapConcat(ScopeAttributes, ResourceAttributes, Attributes)) AS AttributesHash
+          name: 'Source',
+          sql: chSql`
+                SELECT
+                  *,
+                  cityHash64(mapConcat(ScopeAttributes, ResourceAttributes, Attributes)) AS AttributesHash,
+                  any(AttributesHash) OVER (ORDER BY AttributesHash, TimeUnix ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING) AS PrevAttributesHash,
+                  any(Value) OVER (ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING) AS PrevValue,
+                  IF(AggregationTemporality = 1, Value,
+                    IF(Value - PrevValue < 0 AND AttributesHash = PrevAttributesHash, Value,
+                      IF(AttributesHash != PrevAttributesHash, 0, Value - PrevValue))) AS Rate
                 FROM ${renderFrom({ from: { ...from, tableName: metricTables[MetricsDataType.Sum] } })}
-                WHERE MetricName = '${metricName}'
-                ORDER BY AttributesHash, TimeUnix ASC
-            ) `,
+                WHERE MetricName = '${metricName}'`,
```

Comment:
I think we want to do the time range filtering here so db won't scan the whole table

Reply:
I agree but this is going to be a bit more complex since it can't simply translate the range into where conditions. It's going to need to also figure out how to select back one bucket in order to compute the correct values.

Reply:
not sure if its helpful. what I did for the gauge is to reuse the `renderWhere` (https://github.com/hyperdxio/hyperdx/blob/14ad0ff6d5b3b1622b1638b8270b34f45cbf9336/packages/common-utils/src/renderChartConfig.ts#L815-L830) which should handle the custom where clause and timestamp filtering

Reply:
I did try that and then the test got different min/max values because it was missing the leading 0 values. I wonder if simply adjusting the Date object is enough. 🤔

Reply:
oh...maybe its because the time filtering doens't cover the previous window

Reply:
afaik, in v1 there is a time range hack https://github.com/DeploySentinel/hyperdx-ee/blob/e41006508e01e47192845d99043b87ecff97a5bd/packages/api/src/clickhouse/index.ts#L1366-L1372

Reply:
I did something similar but also aligned the range to the start of the interval in the conditional.